File: solutions/c/resistor-color/1/resistor_color.c
#include "resistor_color.h"

static const resistor_band_t band_array[] = {
    BLACK,BROWN,RED,ORANGE,YELLOW,GREEN,BLUE,VIOLET,GREY,WHITE
};

int color_code(resistor_band_t color){
    return (int) color;
}

const resistor_band_t* colors(){
    return band_array;
}

File: solutions/java/resistor-color/1/src/main/java/ResistorColor.java
class ResistorColor {
    int colorCode(String color) {
        color = color.toLowerCase();
        for(int i = 0;i<10;i++){
            if (colors()[i].equals(color)){
                return i;
            }
        }
        return -1;
    }

    String[] colors() {
        return new String[]{"black","brown","red","orange","yellow","green","blue","violet","grey","white"};
    }
}
